File: frontend/src/lib/tasksApi.js
import { RequestJson } from "./apiClient.js";

const TasksBase = "/integrations/google/tasks";

export const FetchTaskLists = async () => RequestJson(`${TasksBase}/lists`);

export const FetchTasks = async (listKey, options = {}) => {
  const refresh = options.Refresh ? "&refresh=1" : "";
  return RequestJson(`${TasksBase}?list_key=${encodeURIComponent(listKey)}${refresh}`);
};

export const CreateTask = async (payload) =>
  RequestJson(TasksBase, {
    method: "POST",
    body: JSON.stringify(payload)
  });

export const UpdateTask = async (taskId, payload) =>
  RequestJson(`${TasksBase}/${taskId}`, {
    method: "PUT",
    body: JSON.stringify(payload)
  });

export const DeleteTask = async (taskId, listKey) =>
  RequestJson(`${TasksBase}/${taskId}?list_key=${encodeURIComponent(listKey)}`, {
    method: "DELETE"
  });

export const FetchTaskSettings = async () => RequestJson("/tasks/settings");

export const UpdateTaskSettings = async (payload) =>
  RequestJson("/tasks/settings", {
    method: "PUT",
    body: JSON.stringify(payload)
  });

export const RunTaskOverdueNotifications = async (force = false) => {
  const query = force ? "?force=1" : "";
  return RequestJson(`/integrations/google/tasks/overdue/run${query}`, {
    method: "POST"
  });
};

File: apps/ios/EverdayIOS/EverdayIOS/Services/TasksApi.swift
import Foundation

enum TasksApi {
    static func fetchLists() async throws -> TaskListsResponse {
        try await ApiClient.shared.request(
            path: "integrations/google/tasks/lists",
            requiresAuth: true
        )
    }

    static func fetchTasks(listKey: String, refresh: Bool = false) async throws -> TaskListResponse {
        let escapedKey = listKey.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) ?? listKey
        let refreshValue = refresh ? "&refresh=1" : ""
        return try await ApiClient.shared.request(
            path: "integrations/google/tasks?list_key=\(escapedKey)\(refreshValue)",
            requiresAuth: true
        )
    }

    static func createTask(_ payload: TaskCreate) async throws -> TaskItem {
        try await ApiClient.shared.request(
            path: "integrations/google/tasks",
            method: "POST",
            body: payload,
            requiresAuth: true
        )
    }

    static func updateTask(taskId: String, payload: TaskUpdate) async throws -> TaskItem {
        try await ApiClient.shared.request(
            path: "integrations/google/tasks/\(taskId)",
            method: "PUT",
            body: payload,
            requiresAuth: true
        )
    }

    static func deleteTask(taskId: String, listKey: String) async throws {
        let escapedKey = listKey.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) ?? listKey
        try await ApiClient.shared.requestVoid(
            path: "integrations/google/tasks/\(taskId)?list_key=\(escapedKey)",
            method: "DELETE",
            requiresAuth: true
        )
    }
}
